python/solutii/dragos_silion/unic.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+# *-* coding: UTF-8 *-*
+
+"""În laboratorul lui Tuxy toți cercetătorii au asignat un id
+de utilizator.
+
+Pentru fiecare cercetător se salvează într-o listă de fiecare
+dată când a deschis ușa (fie pentru a intra, fie pentru a ieși).
+
+Tuxy suspectează că cineva rămâne tot timpul după program și
+ar dori să scrie un script care să îi verifice teoria, dar
+nu a reușit pentru că algoritmul său era prea costisitor pentru
+sistem.
+
+Cerințe:
+    I. Găsește cercetătorul ce stă peste program după o singură
+    parcurgere a listei
+    II. Găsește cercetătorul ce stă peste program după o singură
+    parcurgere a listei și fără a aloca memorie suplimentară.
+"""
+# pylint: disable=unused-argument
+
+
+def gaseste_unic(istoric):
+    """Găsește elementul unic.
+
+    Funcția primește o listă cu elemente numerice și trebuie
+    să returneze elementul care nu este duplicat.
+
+    Exemple:
+        1 2 3 2 1 - 3
+        1 1 1 2 2 - 1
+    """
+    total_xor = 0
```

Comment:
Mulțumesc de feedback!
Legat de fișierul ăla din folderul exerciții, cred că mi-a scăpat. Scuze.

2016-03-27 19:33 GMT+03:00 Alexandru Coman <notifications@github.com>:

> In python/solutii/dragos_silion/unic.py
> <https://github.com/alexandrucoman/labs/pull/127#discussion_r57531209>:
>
> > +    parcurgere a listei și fără a aloca memorie suplimentară.
> > +"""
> > +# pylint: disable=unused-argument
> > +
> > +
> > +def gaseste_unic(istoric):
> > +    """Găsește elementul unic.
> > +
> > +    Funcția primește o listă cu elemente numerice și trebuie
> > +    să returneze elementul care nu este duplicat.
> > +
> > +    Exemple:
> > +        1 2 3 2 1 - 3
> > +        1 1 1 2 2 - 1
> > +    """
> > +    total_xor = 0
>
> Folosești memorie suplimentară.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/alexandrucoman/labs/pull/127/files/46a4e1f5cd08b047e56410008f1fe26dfe134726#r57531209>
>